src/Adapter/Geolocation/GeolocationOptionsConfiguration.php

```diff
@@ -0,0 +1,86 @@
+<?php
+/**
+ * 2007-2018 PrestaShop
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Adapter\Geolocation;
+
+use PrestaShop\PrestaShop\Adapter\Configuration;
+use PrestaShop\PrestaShop\Core\Configuration\DataConfigurationInterface;
+
+/**
+ * Class GeolocationOptionsConfiguration is responsible for configuring geolocation options data
+ */
+final class GeolocationOptionsConfiguration implements DataConfigurationInterface
+{
+    /**
+     * @var Configuration
+     */
+    private $configuration;
+
+    /**
+     * @param Configuration $configuration
+     */
+    public function __construct(Configuration $configuration)
```

Comment:
no problem, ill do that. :+1: 

but why cant we extend current interface with new methods? Is there particular reason why `AdvancedConfigurationInterface` would be better  `ConfigurationInterface`?

Reply:
because if someone is already using the current interface, once we will update it his code will break :D

Reply:
oh yeah. but maybe not deprecating `ConfigurationInterface` is also an option, since we dont always need these 'advanced' methods? anyway, we can discuss it on another PR. :+1:

Reply:
Yes, maybe... it's only a PHPDoc tag, don't add an error_log (you cant do it in interfaces anyway)